src/features/Courses/CoursesDetailPage/index.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useEffect } from 'react';
+import { Link, useParams } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { Container, Pagination } from '@edx/paragon';
+
+import { fetchClassesData } from 'features/Classes/data/thunks';
+import { fetchCoursesData } from 'features/Courses/data/thunks';
+import { updateCurrentPage } from 'features/Courses/data/slice';
+import CourseDetailTable from 'features/Courses/CourseDetailTable';
+import { fetchClassesDataSuccess } from 'features/Classes/data/slice';
+import { initialPage } from 'features/constants';
+
+import 'features/Courses/CoursesDetailPage/index.scss';
+
+const CoursesDetailPage = () => {
+  const dispatch = useDispatch();
+  const { classId } = useParams();
+
+  const [currentPage, setCurrentPage] = useState(initialPage);
+
+  const defaultCourseInfo = {
+    numberOfStudents: '-',
+    numberOfPendingStudents: '-',
+  };
+
+  const courseInfo = useSelector((state) => state.courses.table.data)
+    .find((course) => course?.masterCourseName === classId) || defaultCourseInfo;
+  const institution = useSelector((state) => state.main.selectedInstitution);
+  const classes = useSelector((state) => state.classes.table);
+
+  const handlePagination = (targetPage) => {
+    setCurrentPage(targetPage);
+    dispatch(updateCurrentPage(targetPage));
+  };
+
+  useEffect(() => {
+    if (institution.id) {
+      dispatch(fetchClassesData(institution.id, 1, classId));
```

Comment:
```suggestion
      dispatch(fetchClassesData(institution.id, FirstPage, classId));
```
Same comment as before. @01001110J